### ISSUE TITLE
Build from source fails in release 1.3.2 (ubuntu 18.04)

### ISSUE THREAD

**Issue by binarymac** [zcoinofficial/ccminer#59]
Build from source worked flawlessly in previous versions. Now it fails with error:
```
make[2]: 
nvcc -gencode=arch=compute_61,code=\"sm_61,compute_61\" -gencode=arch=compute_52,code=\"sm_52,compute_52\" -gencode=arch=compute_75,code=\"sm_75,compute_75\" -I/usr/local/cuda/include -I. -I./compat/bos-jansson -std=c++11 --ptxas-options="-v" --maxrregcount=128 -o cuda_mtp/mtp.o -c cuda_mtp/mtp.cu
/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(11265): error: identifier "__builtin_ia32_scalefsd_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(11274): error: identifier "__builtin_ia32_scalefss_round" is undefined

```
**Build environment:** (Cuda 10.2)
`nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Wed_Oct_23_19:24:38_PDT_2019
Cuda compilation tools, release 10.2, V10.2.89`

`gcc --version
gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.`

**Comment by binarymac** [comment]
Thank you very much for your quick reply and the great job you're doing with ccminer!

> I would suggest to do a clean compilation with a git clone
I did it like you suggested (double checked on make clean etc.) but unfortunately it didn't help.

> Did you proceed to some upgrade of gcc version
I did an upgrade of the base system, but didn't check if gcc was upgraded.

> Can you try to check with an older version of ccminer source if it still compile.
I tried with 1.3.1 and 1.3.0 and it doesn't compile either. So it has to be some problem with my system configuration and not with ccminer.

In the past, when I did any OS and Cuda upgrades and the ccminer build broke, I still managed to get the compilation of ccminer working (analysing the error messages). But now I'm stuck, because I don't find anything helpful. The only thing related I found is maybe https://gcc.gnu.org/bugzilla//show_bug.cgi?id=84191 

I have tried to use your released 10.2 binary but it has a dependency to libcurl.so.3 and doesn't start on my system. I have tried to copy and ln libcurl.so.4 but have not been succesful yet. It's a pitty because my previously compiled version of ccminer is working very well (besides the high cpu usage). :-(

**Comment by djm34** [comment]
may-be try to reinstal cuda. Cuda is using internally gcc so the problem might be there...

**Comment by djm34** [comment]
may-be try to install libcurl4-openssl-dev, that's what is used to compile my version

**Comment by binarymac** [comment]
It's working now. I purged the whole nvidia and cuda installation, reinstalled everything and it still didn't work. Then I remembered that in the bugzilla report, that I mentioned above, the closing comment was:
> Fixed in GCC 8.

So I did a parallel install of GCC 8 alongside GCC 7 (default for ubuntu 18.04), set GCC 8 as the default compiler and the compilation worked. Error is gonel and the cpu usage of ccminer is now between 0.3% and 0.7%. 
Thank you again for your support!
